src/sagemaker/lineage/artifact.py

```diff
@@ -147,33 +159,33 @@ def downstream_trials(self, sagemaker_session=None):
         get_module("smexperiments")
         from smexperiments import trial_component, search_expression
 
-        max_search_by_arn = 60
+        max_search_by_arn: int = 60
         num_search_batches = math.ceil(len(trial_component_arns) % max_search_by_arn)
-        trial_components = []
+        trial_components: list = []
 
         sagemaker_session = sagemaker_session or _utils.default_session()
         sagemaker_client = sagemaker_session.sagemaker_client
 
         for i in range(num_search_batches):
-            start = i * max_search_by_arn
-            end = start + max_search_by_arn
-            arn_batch = trial_component_arns[start:end]
-            se = self._get_search_expression(arn_batch, search_expression)
-            search_result = trial_component.TrialComponent.search(
+            start: int = i * max_search_by_arn
+            end: int = start + max_search_by_arn
+            arn_batch: list = trial_component_arns[start:end]
+            se: Any = self._get_search_expression(arn_batch, search_expression)
+            search_result: Any = trial_component.TrialComponent.search(
```

Comment:
its inside of the method, are those annotation really needed?

Reply:
It is more convenient to know what the variables' data types are , even inside the method, thus the annotation.